internal/controllers/provisioningrequest_clusterconfig.go

```diff
@@ -248,6 +256,72 @@ func (t *provisioningRequestReconcilerTask) finalizeProvisioningIfComplete(ctx c
 	return nil
 }
 
+// addClusterTemplateLabels adds the clustertemplates.o2ims.provisioning.oran.org/templateId on the
+// ManagedCluster and Agent objects that are associated to the current ProvisioningRequest.
+func (t *provisioningRequestReconcilerTask) addClusterTemplateLabels(ctx context.Context, mcl *clusterv1.ManagedCluster) error {
+	// Get the ClusterTemplate used by the current ProvisioningRequest.
+	oranct, err := t.object.GetClusterTemplateRef(ctx, t.client)
+	if err != nil {
+		return fmt.Errorf("failed to get ClusterTemplate: %w", err)
+	}
+
+	// Add the clustertemplates.o2ims.provisioning.oran.org/templateId label to the ManagedCluster
+	// associated to the current ProvisioningRequest.
+	err = t.setLabelValue(ctx, mcl, utils.ClusterTemplateArtifactsLabel, oranct.Spec.TemplateID)
+	if err != nil {
+		return err
+	}
+
+	// Add the clustertemplates.o2ims.provisioning.oran.org/templateIds label to the Agent
+	// associated to the current ProvisioningRequest.
+	agents := &assistedservicev1beta1.AgentList{}
+	listOpts := []client.ListOption{
+		client.MatchingLabels{
+			"agent-install.openshift.io/clusterdeployment-namespace": mcl.Name,
+		},
+		client.InNamespace(mcl.Name),
+	}
+
+	err = t.client.List(ctx, agents, listOpts...)
+	if err != nil {
+		return fmt.Errorf("failed to list Agents in the %s namespace: %w", mcl.Name, err)
+	}
+
+	if len(agents.Items) > 1 {
+		return fmt.Errorf("more than one Agent in the %s namespace", mcl.Name)
```

Comment:
I may be misreading this, but isn't `> 1` a possibility in a MNO setup?  In which case we'd want to label all of them?

Reply:
each agent (node instance) would be in its own NS I believe

Reply:
What I have seen before is that Agent for each node is under the cluster NS. Same for sno + worker.

Reply:
I remember seeing the same on 3 node spoke clusters. I will check in the QE labs for 3 node and standard spoke clusters and I'll open a following PR to properly address those scenarios.